stay in touch (milestone 6)

che30 opened the pull request:
In this project we : 
- Added a friendship model and it's associations
- Added buttons to add, decline and accept frienship on users index and show page
- Add integration test
- Add unit test using Rspec

descholar-ceo commented:
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

descholar-ceo commented on app/helpers/friendships_helper.rb:
```diff
@@ -0,0 +1,43 @@
+# rubocop:disable Layout/LineLength
+
+# rubocop:disable Style/GuardClause
+# rubocop:disable Lint/AssignmentInCondition
```
[required] You should try to fix rubocop lint errors instead of disabling them

descholar-ceo commented on app/models/user.rb:
```diff
@@ -1,3 +1,6 @@
+
+# rubocop:disable Lint/ShadowingOuterLocalVariable
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
[required] It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

descholar-ceo reviewed (changes requested):
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @che30 and @Fanger53  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have implemented carefully almost all of the requirements of this milestone
:+1: I liked the `CANCEL` button you added when I send a friend request to another user, so after some time I realise that I clicked the `send friend request` button mistakenly, then I will be able to cancel that action, => Cudos :muscle:  about that extra effort, it's really commendable, keep it up :muscle:  
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/auth/sign-in/?next=/library/fast-track-curriculum/69047/path/step/49736080/)) => Currently this requirements is not implemented as it stated in project specifications, a user is getting all of the posts from anyone, whether theirs, their friend's and posts from anyone else are being displayed on their timeline, please fix this to show only the post from logged in user and post from their friend's only.
- [ ] You are missing tests, you have to write tests (bot unit and integration tests) using rspec
- [ ] Remove commented codes: Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] Avoid disabling rubocop lint errors, this will help you train yourself to write beautiful, clean and maintainable codes
- [ ] You have some logics in views: There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from `Pathwright`
Resources
    - [ ] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb) => To help you unerstand what to change: Make sure that there are no if conditions, or loops in views

Resources that can help you: 
:point_right: [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
:point_right: [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [ ] Implement all changes I suggested in inline comments in your codes

# [Optional changes] :yellow_circle: 

- [ ] It would be very nice if you add the currently logged in user in navbar, so that I will be able to see my username in navbar when i am logged in
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

che30 commented:
Added rspec tests
Added user  logic for user logic and his timeline
removed comment
Tried to seperate the logic completely but I can't up until now Your help will be welcomed

che30 commented:
Added rspec tests
Added user  logic for user logic and his timeline
removed comment
Tried to seperate the logic completely but I can't up until now Your help will be welcomed

jubaan reviewed (changes requested):
# Status: Changes Required :recycle: 

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [ ] There is an issue when User declines another User friendship invitation
![image](https://user-images.githubusercontent.com/25774518/105046127-4fdfe200-5a2e-11eb-8d7d-33c6f3fb50bd.png)

- [ ] Timeline is only showing the current user post, and it should show also the post from the accepted friends.
- [ ] Remove all logic application logic from the views, and create and use rails custom helpers. Example:
```ruby
# ./app/views/comments/_comment.html/erb

# BAD
<span> <%= comment.created_at.strftime("%Y/%m/%d") %> </span>


# GOOD
  <span> <%= comment_creation_time(comment) %> </span>

# ./app/helpers/comments_helper.rb
def comment_creation_time(comment)
    comment.created_at.strftime("%Y/%m/%d")
end
```

- [ ] Rubocop should be enabled, as asked by the previous reviewer. files: friendship_helper and friendship_controller


## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

MahmoudBakr23 reviewed (changes requested):
All changes implemented as required

mahmoud717 reviewed (approved):
### STATUS: APPROVED ✔️ ✔️ 

**hi @che30**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**
